### ISSUE TITLE
[public-1.13.5] Sound Issues (ambient_generic, Soundscapes and music_manager)

### ISSUE THREAD

**Issue by Fersigal** [nmrih/source-game#1482]
Version: 1.13.5 Official Game Version
**What happens:** 
- Sounds from env_soundscape seems to be broken
- Sounds from music_manager seem to be very infrequent or be not playing at all (even on maps like cleopas)
This probably happens, because of the new changes with the sound queues (ambient_fmod and what not has been added, but random sound occurrences could be dating back to a couple of patches ago)
**When does it happen:**
- When entering a new area or exiting a new area where the env_soundscape cuts off or when you reenter a certain area
**Steps to reproduce:**
1. Be in an area with a env_soundscape
2. Stay in the area or enter and reexit the area and enjoy the same sound being played over and over again
**Video Examples:**
1.https://youtu.be/rBb3QCkOOqk
2.https://youtu.be/2pOBl4z0awg

System Information:

> Computer Information:
> Manufacturer: Gigabyte Technology Co., Ltd.
> Model: Z390 GAMING X
> Form Factor: Desktop
> No Touch Input Detected
> Processor Information:
> CPU Vendor: GenuineIntel
> CPU Brand: Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz
> CPU Family: 0x6
> CPU Model: 0x9e
> CPU Stepping: 0xd
> CPU Type: 0x0
> Speed: 3600 MHz
> 8 logical processors
> 8 physical processors
> Hyper-threading: Unsupported
> FCMOV: Supported
> SSE2: Supported
> SSE3: Supported
> SSSE3: Supported
> SSE4a: Unsupported
> SSE41: Supported
> SSE42: Supported
> AES: Supported
> AVX: Supported
> AVX2: Supported
> AVX512F: Unsupported
> AVX512PF: Unsupported
> AVX512ER: Unsupported
> AVX512CD: Unsupported
> AVX512VNNI: Unsupported
> SHA: Unsupported
> CMPXCHG16B: Supported
> LAHF/SAHF: Supported
> PrefetchW: Unsupported
> Operating System Version:
> Windows 10 (64 bit)
> NTFS: Supported
> Crypto Provider Codes: Supported 311 0x0 0x0 0x0
> Video Card:
> Driver: NVIDIA GeForce RTX 2080
> DirectX Driver Name: nvldumd.dll
> Driver Version: 31.0.15.3623
> DirectX Driver Version: 31.0.15.3623
> Driver Date: 6 8 2023
> OpenGL Version: 4.6
> Desktop Color Depth: 32 bits per pixel
> Monitor Refresh Rate: 60 Hz
> DirectX Card: NVIDIA GeForce RTX 2080
> VendorID: 0x10de
> DeviceID: 0x1e87
> Revision: 0xa1
> Number of Monitors: 2
> Number of Logical Video Cards: 2
> No SLI or Crossfire Detected
> Primary Display Resolution: 1920 x 1080
> Desktop Resolution: 3840 x 1200
> Primary Display Size: 20.51" x 11.54" (23.50" diag), 52.1cm x 29.3cm (59.7cm diag)
> Primary Bus: PCI Express 16x
> Primary VRAM: 8191 MB
> Supported MSAA Modes: 2x 4x 8x
> Sound card:
> Audio device: Luidsprekers (2- USB PnP Sound
> Memory:
> RAM: 16315 Mb
> VR Hardware:
> VR Headset: None detected
> Miscellaneous:
> UI Language: English
> Media Type: Undetermined
> Total Hard Disk Space Available: 953223 MB
> Largest Free Hard Disk Block: 373126 MB
> OS Install Date: dec 05 2020
> Game Controller: None detected
> MAC Address hash: 8391f5807b10cf5d39f84c973d72e300c4fba65f
> Storage:
> Disk serial number hash: 37bf6893
> Number of SSDs: 1
> SSD sizes: 1000G
> Number of HDDs: 0
> Number of removable drives: 0

**Comment by felis-catus** [comment]
Resolved issues:
1. Room tone cutting short is caused by an outdated sound cache, it's fixed in 1.13.6 by shipping a rebuilt one (we resolved an issue that prevented us from rebuilding caches), current workaround is deleting `nmrih.vpk.sound.cache` in the `nmrih` directory.
2. `ambient_generic` + `ambient_fmod` audible sound level issues were resolved in src-r7550.
3. Adding support for `ambient_generic` MP3 looping broke maps that erroneously set spawnflags to loop on MP3 sounds, in 1.13.6 mappers must opt-in to make use of the feature.
4. `snd_restart` behavior is also fixed in 1.13.6, and can also restart FMOD ambient sounds.

Need investigation:
1. `music_manager` didn't receive updates in 1.13.5, but FMOD sound system regressions are possible.
